### DISCUSSION TITLE
What is the difference between enr.results and enr.res2d?

### DISCUSSION THREAD

**Discussion by zhoutianrui-tongji** [zqfang/GSEApy#199] [Q&A] [answered]
I found the following code while looking at GSEAPY's documentation.
`# categorical scatterplot
ax = barplot(enr.results,
              column="Adjusted P-value",
              group='Gene_set', # set group, so you could do a multi-sample/library comparsion
              size=10,
              top_term=5,
              figsize=(3,5),
              color=['darkred', 'darkblue'] # set colors for group
             )`
and
`ax = barplot(enr.res2d,title='KEGG_2021_Human', figsize=(4, 5), color='darkred')`
**I don't quite understand the difference between these two, especially why we need to use enr.results and enr.res2d. What is the difference between these two?**

**Comment by zqfang** [comment, marked as answer]
enr.res2d only stores the last gene_set while enr.results contains all gene_sets 

`.res2d` is a common interface in all modules in gseapy
